src/blob/index/core.rs

```diff
@@ -93,16 +94,56 @@ impl<FileIndex: FileIndexTrait> IndexStruct<FileIndex> {
         self.range_filter.clear();
     }
 
-    pub(crate) fn check_filters_key(&self, key: &[u8]) -> FilterResult {
+    pub fn offload_filter(&mut self) {
+        if self.on_disk() {
+            self.bloom_filter.offload_from_memory();
+        }
+    }
+
+    pub(crate) async fn check_filters_key(&self, key: &[u8]) -> FilterResult {
         if !self.range_filter.contains(key) {
             FilterResult::NotContains
-        } else if self.params.bloom_is_on && !self.bloom_filter.contains(key) {
+        } else if self.params.bloom_is_on
+            && matches!(self.check_bloom_key(key).await, Ok(Some(false)))
```

Comment:
It works fine. But we need to investigate errors somehow. And errors would be just dropped here. Maybe it makes sense to return `Result<FilterResult>` here